include/eve/detail/function/broadcast_group.hpp

```diff
@@ -0,0 +1,91 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/meta.hpp>
+#include <eve/detail/overload.hpp>
+#include <eve/concept/vectorized.hpp>
+#include <eve/pattern.hpp>
+#include <bit>
+
+namespace eve
+{
+  EVE_MAKE_CALLABLE(broadcast_group_, broadcast_group);
+
+  //================================================================================================
+  // Premade pattern generator
+  template<std::ptrdiff_t G, std::ptrdiff_t I, std::ptrdiff_t N>
+  inline constexpr auto broadcast_group_n = fix_pattern<N>( [](auto i, auto) { return I*G + i%G; } );
+
+  //================================================================================================
+  // Classify a pattern as a bcstg
+  namespace detail
+  {
+    // Find the proper broadcast group shape. IF none is found, returns {-1,-1}
+    template<std::ptrdiff_t... Is> constexpr inline auto bcstg_check()
```

Comment:
It seems like it's entirely operates on std::ptrdiff_t

Why not just have a constexpr function from std::array<std::ptrdiff_t>?
Then you can use normal loops and don't need anything.

Reply:
This is a remnant of an ancient design? INdeed at this point I can operates on a array I guess

Reply:
I didn't mean to change the interface. But you have some for_ thing and I don't think you need that here

Reply:
Well, operating on array there is OK